### ISSUE TITLE
[REQUEST] Use GPU on Mac M1

### ISSUE THREAD

**Issue by mindacrobatic** [takuseno/d3rlpy#293]
**Is your feature request related to a problem? Please describe.**
I want to train models with d3rlpy on my Mac M1, but currently I don't see how i can use the builtin GPU of my laptop (MPS).

**Describe the solution you'd like**
I would like to use d3rlpy with MPS just like it works with CUDA.

**Comment by takuseno** [comment]
@mindacrobatic Thanks for the issue. Currently, this is not supported. However, the upcoming next version is supporting this. The documentation is not ready yet, but you can actually test this with [nightly](https://github.com/takuseno/d3rlpy/tree/nightly) branch.
```
git clone https://github.com/takuseno/d3rlpy
cd d3rlpy
git checkout nightly
pip install -e .
```
`examples` and `reproductions` directories in `nightly` branch are already updated. So you can test M1 Macbook's GPU as follows:
```
import d3rlpy

dataset, env = d3rlpy.dataset.get_pendulum()
cql = d3rlpy.algos.CQLConfig().create(device="mps")
cql.fit(dataset, n_steps=100000, scorers={'environment': d3rlpy.metrics.EnvironmentEvaluator(env)})
```

**Comment by mindacrobatic** [comment]
@takuseno Thank you so much, that is amazing!

The second part of your code didn't run (I think it was complaining about the dataset), but after some adjustments I was able to run it on my offline dataset. Can't wait to have this branch being merged!

**Comment by mindacrobatic** [comment]
@takuseno Takuseno, I was about to close this issue, but then I dropped over something that lets me doubt if everything is fine in this branch.

This is my example code. You can see that I set the device to "mps". During training, I get around 50-60 it/s. If I swap the device to CPU, I get around 400 it/s. That is not how it is supposed to be, right?

```
import d3rlpy
from d3rlpy.algos import DiscreteCQL, DiscreteCQLConfig

config = DiscreteCQLConfig(
    learning_rate=6.0e-05
)

cql = DiscreteCQL(config=config, device="mps")

history = cql.fit(
    dataset, 
    n_steps=52 * 80_000,
    n_steps_per_epoch=52,
    save_interval=1000000
)
```

**Comment by takuseno** [comment]
Thank you for the test. Unfortunately, I don't have Macbooks to test, but it seems that `mps` in PyTorch seems not ready for training yet:
https://github.com/pytorch/pytorch/issues/77799

**Comment by takuseno** [comment]
Feel free to reopen this if you have any further discussion. Please note that the nightly branch has been merged and the latest update `v2.0.2` has been released.
